### PR TITLE
Custom marks

### LINKED ISSUE #27: Create example showing filter/highlight behavior by Query status

### PR DESCRIPTION
**Pull request by samussiah**
Closes #27 

Alternatively visualize queries with modals: http://bl.ocks.org/samussiah/raw/afe3abbce6b574926536bb12292a4e72/.  This method is established in https://github.com/RhoInc/safety-outlier-explorer/tree/query-highlighting-with-modals

### CODE REVIEW

File: src/smallMultiples.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ export default function smallMultiples(id, chart) {
         rangePolygon(this);
 
         // axis tweaks
-        var units = this.current_data[0].values.raw[0][chart.config.unit_col];
+        var units = this.raw_data[0][chart.config.unit_col];
```
Comment (jwildfire) [inline src/smallMultiples.js]: This probably won't work since the `unit_col` typically changes with `measure_col`

Comment (jwildfire) [inline src/smallMultiples.js]: might need something like `raw_data.filter(d=>give me the current measure)[0][chart.config.unit_col]`

Comment (jwildfire) [inline src/smallMultiples.js]: we should really clean up the data paradigm in webcharts a bit ... let's make time for that after these releases go out.

File: src/util/rangePolygon.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ export default function rangePolygon (chart){
             return lbornrhi !== 'NA' ? chart.y(+lbornrhi) : 0
         });
 
-    var dRow = chart.filtered_data[0];
+    var dRow = chart.raw_data[0];
```
Comment (jwildfire) [inline src/util/rangePolygon.js]: Need to only look at the currently selected measure here. see comment above for possible implementation.

File: src/smallMultiples.js (hunk 1 above)
Comment (jwildfire) [inline src/smallMultiples.js]: For the moment, maybe we should create/maintain a `chart.measure_data` in this renderer. Seems like that would deal with most of this nonsense in a fairly elegant way ... just throw something like this in to `onInit()` and `onDraw()`:

```
var current_measure = {go get it from the control}
chart.measure_data = raw_data.filter(d=>d[chart.config.measure_col]==current_measure)
```

File: src/default-settings.js [hunk 3]
```diff
@@ -48,25 +49,30 @@ const settings = {
...
     settings.marks[0].tooltip = `[${settings.id_col}]`;
+
     settings.marks[1].per = [
         settings.id_col, 
         settings.measure_col,
         settings.time_cols[0],
         settings.value_col
     ];
     settings.marks[1].tooltip = `[${settings.id_col}]:  [${settings.value_col}] [${settings.unit_col}] at ${settings.x.column} = [${settings.x.column}]`;
+
```
Comment (jwildfire) [inline src/default-settings.js]: Yessiree. Like this a lot.

**Comment by jwildfire** [comment]
Note: created (an issue in webcharts)[https://github.com/RhoInc/Webcharts/issues/31] to consider adding support for modals.

File: src/smallMultiples.js (hunk 1 above)
Comment (samussiah) [inline src/smallMultiples.js]: @jwildfire `this.raw_data` in this case refers to an individual multiple's data, which is already filtered on a single measure.  The reason I switched from `this.current_data` eludes me but it is simpler to pull a datum out of `this.raw_data`.

File: src/util/rangePolygon.js (hunk 2 above)
Comment (samussiah) [inline src/util/rangePolygon.js]: Similarly, `chart.raw_data` refers to the multiple's data, which is already filtered on a single measure.

File: src/smallMultiples.js (hunk 1 above)
Comment (samussiah) [inline src/smallMultiples.js]: Just kidding, you're absolutely right.  Will update to `this.filtered_data` in a new branch.

File: src/util/rangePolygon.js (hunk 2 above)
Comment (samussiah) [inline src/util/rangePolygon.js]: Just kidding, you're absolutely right.  Will update to `this.filtered_data` in a new branch.

**Pull request merged**